src/spi.rs

```diff
@@ -372,7 +372,7 @@ where
         } else if sr.crcerr().is_no_match() {
             nb::Error::Other(Error::Crc)
         } else if sr.txe().is_empty() {
-            let write_ptr = core::ptr::addr_of!(self.spi.dr) as *mut Word;
+            let write_ptr = self.spi.dr().as_ptr() as *mut Word;
             // SAFETY: Write to register owned by this Spi struct
             unsafe { core::ptr::write_volatile(write_ptr, word) };
```

Comment:
I'd recommend to use `dr` register for `Word = u16` and `dr8` register for `Word = u8` so you don't need for casting:
```rust
self.spi.dr8().write(|w| dr().set(word));
```
Same for reading.

Reply:
Something like 9cb658e?

Reply:
Yes